src/receiving.rs

```diff
@@ -323,14 +346,15 @@ impl Receiver {
     ///
     /// * One of the public keys to scan can't be parsed into a valid x-only public key.
     /// * An error occurs during elliptic curve computation. This may happen if a sender is being malicious. (?)
-    pub fn scan_transaction_with_labels(
+    pub fn scan_transaction(
         &self,
         ecdh_shared_secret: &PublicKey,
         pubkeys_to_check: Vec<XOnlyPublicKey>,
-    ) -> Result<HashMap<Label, HashMap<XOnlyPublicKey, Scalar>>> {
+        with_labels: bool,
```

Comment:
I agree with the idea of dropping the other `scan_transaction` and just using this one. I'm also leaning towards dropping this 'with_labels' argument, and just always scan with labels (including the change label). The BIP recommends this too: "every receiving silent payments wallet should at least scan for the change label when recovering from backup in order to ensure maximum cross-compatibility". 

So for simplicity I would say to always scan for change label.

Reply:
Performance would be the only reason for not always scanning for label. The only way to know for sure would be to run benchmarks. You're right that leaving the option not to scan the change label would not be compliant with the bip. 
Maybe the safest is to drop the bool for now and always scan, if in practice there seems to be issues with that or use cases for not scanning change labels we will always be able to put it back later